src/Objects/MonitorManager.vala

```diff
@@ -237,9 +250,11 @@ public class Display.MonitorManager : GLib.Object {
                 virtual_monitor.primary = false;
                 virtual_monitor.monitors.add (monitor);
                 virtual_monitor.scale = virtual_monitors[0].scale;
-                add_virtual_monitor (virtual_monitor);
+                virtual_monitors.add (virtual_monitor);
             }
         }
+
+        layout_manager.arrange_monitors (virtual_monitors);
```

Comment:
disable_clone_mode is already re-arranging the monitors, so we only need to save it here.
```suggestion
        if (!is_mirrored) {
            layout_manager.save_layouts (virtual_monitors);
        }
```